src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java

```diff
@@ -356,6 +482,96 @@ public boolean configure(StaplerRequest req, JSONObject json)
 		}
 	}
 
+	/**
+	 * Checks if there are enough resources available to satisfy the requirements specified
+	 * within requiredResources and returns the necessary available resources.
+	 * If not enough resources are available, returns null.
+	 */
+	public synchronized List<LockableResource> checkResourcesAvailability(LockableResourcesStruct requiredResources,
+			@Nullable PrintStream logger, @Nullable List<String> lockedResourcesAboutToBeUnlocked) {
+		// get possible resources
+		int requiredAmount = 0; // 0 means all
+		List<LockableResource> candidates = new ArrayList<LockableResource>();
+		if (requiredResources.label != null && requiredResources.label.isEmpty()) {
+			candidates = requiredResources.required;
+		} else {
+			candidates = getResourcesWithLabel(requiredResources.label, null);
+			if (requiredResources.requiredNumber != null) {
+				try {
+					requiredAmount = Integer.parseInt(requiredResources.requiredNumber);
+				} catch (NumberFormatException e) {
+					requiredAmount = 0;
+				}
+			}
+		}
+
+		if (requiredAmount == 0) {
+			requiredAmount = candidates.size();
+		}
+
+		// start with an empty set of selected resources
+		List<LockableResource> selected = new ArrayList<LockableResource>();
+
+		// some resources might be already locked, but will be freeed.
+		// Determine if these resources can be reused
+		if (lockedResourcesAboutToBeUnlocked != null) {
+			for (LockableResource candidate : candidates) {
+				if (lockedResourcesAboutToBeUnlocked.contains(candidate.getName())) {
+					selected.add(candidate);
+				}
+			}
+			// if none of the currently locked resources can be reussed,
+			// this context is not suitable to be continued with
+			if (selected.size() == 0) {
+				return null;
+			}
+		}
+
+		for (LockableResource rs : candidates) {
+			if (selected.size() >= requiredAmount) {
+				break;
+			}
+			if (!rs.isReserved() && !rs.isLocked()) {
+				selected.add(rs);
+			}
+		}
+
+		if (selected.size() < requiredAmount) {
+			if (logger != null) {
```

Comment:
This is the fix over #26 (move this check inside the previous one so `return null` is actually reached even when we have no logger).